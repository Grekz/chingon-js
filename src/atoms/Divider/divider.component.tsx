import styled from '@emotion/styled'
import * as React from 'react'

interface IDividerProps {
  color?: string
}

type DividerType = React.SFC<IDividerProps>

const Divider = (props: IDividerProps) => <DividerWrapper {...props} />

const DividerWrapper: DividerType = styled.hr`
  color: ${({ color }) => color};
`

export default Divider
