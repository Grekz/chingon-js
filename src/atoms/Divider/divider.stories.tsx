import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Divider from '.'

storiesOf('Atoms|Divider', module).add('As Default', () => (
  <Divider color="orange" />
))
