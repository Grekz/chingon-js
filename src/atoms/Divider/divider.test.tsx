import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import Divider from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<Divider /> rendering', () => {
  it('should render Divider', () => {
    const wrapper = shallow(<Divider color="orange" />)
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(<Divider color="orange" />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
