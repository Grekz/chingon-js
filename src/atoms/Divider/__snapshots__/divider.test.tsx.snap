// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> rendering should match snapshot 1`] = `
<Styled(hr)
  color="orange"
/>
`;
